File: src/Adapters/AlmacenamientoMemoria.cpp
#include "AlmacenamientoMemoria.h"

void AlmacenamientoMemoria::guardarProducto(const Producto& producto) {
    inventario.push_back(producto);
}

Producto* AlmacenamientoMemoria::buscarProducto(std::string nombre) {
    for (auto& prod : inventario) {
        if (prod.getNombre() == nombre) {
            return &prod;
        }
    }
    return nullptr;
}

std::vector<Producto> AlmacenamientoMemoria::obtenerTodos() {
    return inventario;
}

void AlmacenamientoMemoria::actualizarProducto(const Producto& productoActualizado) {
    for (auto& prod : inventario) {
        if (prod.getId() == productoActualizado.getId()) {
            prod = productoActualizado;
            break;
        }
    }
}

File: frontend/js/adapters/AlmacenamientoMemoria.js
export class AlmacenamientoMemoria {
    constructor() {
        this.inventario = [];
    }
    
    guardarProducto(producto) {
        this.inventario.push(producto);
    }
    
    obtenerTodos() {
        return this.inventario;
    }
    
    buscarProductoPorId(id) {
        return this.inventario.find(p => p.id === id) || null;
    }
    
    actualizarProducto(productoActualizado) {
        const index = this.inventario.findIndex(p => p.id === productoActualizado.id);
        if (index !== -1) {
            this.inventario[index] = productoActualizado;
        }
    }
}